app/models/dependant_income_model.rb

```diff
@@ -5,14 +5,31 @@ class DependantIncomeModel
   include NumberValidatable
 
   ATTRIBUTES = %i[amount frequency].freeze
+  DEPENDANT_INCOME_UPPER_LIMITS = {
+    "every_week" => 78.20,
+    "every_two_weeks" => 156.40,
+    "every_four_weeks" => 312.80,
+    "monthly" => 338.90,
+    "three_months" => 1016.70,
+  }.freeze
 
   attribute :amount, :gbp
   validates :amount, presence: true, numericality: { greater_than: 0, allow_nil: true }
 
   attribute :frequency, :string
   validates :frequency, inclusion: { in: IncomeModel::FREQUENCY_OPTIONS, allow_nil: false }
 
+  validate :income_too_high
+
   def frequency_options
     IncomeModel.income_frequency_options(skip_year: true)
   end
+
+  def income_too_high
+    return unless amount
+
+    if DEPENDANT_INCOME_UPPER_LIMITS[frequency] <= amount
```

Comment:
This will break if the user has submitted an amount (to get past the guard clause on line 29) but not a frequency.

Reply:
💡 silly mistake, will fix!

Reply:
Ooh, there's one other niche issue here. `amount` is of type `gbp`, which has a slightly unusual behaviour: If the user enters a valid number, it will be parsed to a decimal, because functionally it's a number. _But_ if the user enters something invalid, the original value entered will be preserved as a string, so that when the page is re-rendered the original value is pre-populated in the field.

So there is a possibility that `amount` will be a string. What we do in other situations to work around this (e.g. EmploymentForm#net_income_must_be_positive or HousingCostsForm#housing_benefit_does_not_exceed_costs) is a guard clause checking `blah.to_i.positive?` or `errors.none?`, which will ensure if there is some other validation error, this logic won't run. But there may be a neater way.

Reply:
Oh thank you for this spot! Will fix this up too